### DISCUSSION TITLE
Conflicts solving : show mods in conflicts

### DISCUSSION THREAD

**Discussion by mineride** [bcssov/IronyModManager#305] [Ideas]
Hello there,
First off all, thank you for this tools, like it a lot for conflict solving and pretty useful. 

I've got a suggestion for the conflict solver which can make the solving faster i think : The mods sometimes conflicting a lot and increase the number of conflict just because two mods or more have in fact decided to do a brawl inside the game files for each lines of codes possible.

So maybe instead of changing manually show somewhere which mods is conflicting with another mod a lot ( example : expanded stellaris ascension perks and plentiful traditions have a good amount of conflict and really need a patch. So i can see that those two mods are in fact conflicting a lot so i can chose in the mod filter faster)

an example of how it could be made is : "X mod conflict with Y mod with an amount of Z conflicts." with that maybe a button that apply a filter to only see all the conflict between the two mods

i know it's a little weird how I say it (English is not my first language) but it could be usefull to know if we can do the conflict resolving with the chance to see them work together or it's in fact useless because it's need an entire patch that doesn't exist in the first place or we just have to make the correct choice between two mods.

Thank you in advance and have a good day

**Comment by bcssov** [comment]
I'm also not a native English speaker. I do maybe understand you but could you explain a bit more. So far I understand you want to see an option to find if a mod has more than 1 conflict with another mod?

**Reply by mineride** [reply]
well when you try to see the conflict you have the conflict solver that show you all of them under the category of each type of files that they are conflicting (interface, economy, icons,... ect)
However, you can only see which mods is conflicting by looking one by one. so you have to remember if this mods create a lot of problem or not (i like notepad where i write down all conflicts and which one is conflicting but not that much)

I think by putting the option to see how much conflicts the mods X is generating and with what (how much and which mods he conflicts with and how much conflicts for each mods but maybe too much to ask) we can see if the mod is too much to handle (example : making conflict with 2 to 5 mods and 500 conflicts in which category because well, if it's translation, i think we can pass) and that we can't play with it or this have just a few things to modify (example : making conflict with 1 mods and 10 conflicts)

example of what it can be at the best if it's possible : 

mod X have 500 conflicts with : 
- mod A - 200 conflicts (icons (50), ascension perks (100) ,economy (50))
- mod B - 50 conflicts (translations ENG (50))
- mod C - 250 conflicts (megastructures (250))

the numbers in the coma are the conflicts in each category

with that you can know also if you can skip a category entirely (example i've got now gigastructural conflict with itself (wut) and take an entire category which is megastructure for no reason but i've got to check if no other mods is conflicting but i have to check one by one)


it will save time to know if we can do the conflict resolving without obliterating a tons of files by merging all of it by copying all the missing lines and praying it will work but not quite
In addition, it will show to the user if there is a need for a patch or not between two different mods (win - win for users and modders) if it was not detected before and make people make them faster i think (in an utopia world i know).


Hope it was clear enough. My bad ^^. Thanks for the reply

**Comment by bcssov** [comment]
Isn't mod filter enough for this purpose? You toggle a single mod and you can see with what exactly it is conflicting and how many conflicts there are?

**Comment by mineride** [comment]
MMMMMmmmaaybe but i think it could be a little better if you can see directly where they conflicting and with who on a single page and not changing the category everytime even with the Filter on. Like a status report.

Because the mod filter still need you to look with who they are conflicting because the mod filter only look one way and not both.
i take an example i have with mine : 

![image](https://user-images.githubusercontent.com/82579032/139650299-f2c4203b-b193-4281-85c5-520b31eb7565.png)

expanded stellaris traditions activate as the filter, we can see that we have 129 conflicts but with who? we have to look every conflict to know with which mod they are fighting for supremacy over the files and it can take some time.
In addition, we can then clearly see if we can let it as it should be (some conflict are intended for exemple with lots of traditions with the two mod merge but conflict is showing of course but it is intended so we have just to check if the patch is taking the lead)

So if we can have some sort of report on the mod which tells us all the mods they are conflicting with before going for the long work of trying to understand how to avoid doomsday by patching everything correctly, i think it could be useful
However i see the point where the mod filter can make it and i just request a solution just to by some times so i understand it's not necessary.

**Reply by bcssov** [reply]
I'll consider it, I'll leave it here in the discussion for the time being one the guys from discord get here and chime in as well.

**Reply by LordOfLA** [reply]
You should be using an external code editor like Visual Studio Code with the CW Tools extension if you're wanting to make a comprehensive mod interoperability patch.

Irony's purpose is to make it easy to patch individual definition conflicts so you can play a game and have things work mostly as expected.

You should also be making liberal use of ignore rules. My suggested ones for 3.1.2 are attached.

[irony_ignore_rules.txt](https://github.com/bcssov/IronyModManager/files/7455806/irony_ignore_rules.txt)

There's little you need to patch in economic categories anyway, just get the universal modifier patch and ariphaos unofficial patch and either ignore the category outright with an ignore rule, or patch the 4 or 5 entries that really need it.
